src/Adapter/CustomerService/CommandHandler/ForwardCustomerThreadHandler.php

```diff
@@ -0,0 +1,321 @@
+                'Fwd: Customer message',
+                [],
+                'Emails.Subject',
+                $this->context->language->locale
+            ),
+            $params,
+            $command->getEmail()->getValue(),
+            null,
+            $this->context->employee->email,
+            $this->context->employee->firstname . ' ' . $this->context->employee->lastname,
+            null,
+            null,
+            _PS_MAIL_DIR_,
+            true
+        );
+
+        if ($forwardEmailSent) {
+            $customerMessage->message = sprintf(
+                '%s %s \\n %s %s',
+                $this->translator->trans('Message forwarded to', [], 'Admin.Catalog.Feature'),
+                $command->getEmail()->getValue(),
+                $this->translator->trans('Comment:'),
+                $command->getComment()
+            );
+            $customerMessage->add();
+        }
+    }
+
+    /**
+     * @param $message
+     * @param null $id_employee
+     *
+     * @return string
+     */
+    protected function renderMessage($message, $id_employee = null)
+    {
+        $tpl = $this->context->smarty->createTemplate(
+            $this->context->smarty->getTemplateDir(0) . 'message.tpl',
+            $this->context->smarty
+        );
+
+        $contacts = Contact::getContacts($this->context->language->id);
+        $contact_array = [];
+
+        foreach ($contacts as $contact) {
+            $contact_array[$contact['id_contact']] = [
+                'id_contact' => $contact['id_contact'],
+                'name' => $contact['name'],
+            ];
+        }
+
+        $contacts = $contact_array;
+
+        if (!empty($message['id_product'])
+            && empty($message['employee_name'])
+        ) {
+            $id_order_product = Order::getIdOrderProduct(
+                (int) $message['id_customer'],
+                (int) $message['id_product']
+            );
+        }
+
+        $message['date_add'] = Tools::displayDate($message['date_add'], null, true);
+        $message['user_agent'] = strip_tags($message['user_agent']);
+        $message['message'] = preg_replace(
+            '/(https?:\/\/[a-z0-9#%&_=\(\)\.\? \+\-@\/]{6,1000})([\s\n<])/Uui',
+            '<a href="\1">\1</a>\2',
+            html_entity_decode(
+                $message['message'],
+                ENT_QUOTES,
+                'UTF-8'
+            )
+        );
+
+        $isValidOrderId = true;
+        $order = new Order((int) $message['id_order']);
+
+        if (!Validate::isLoadedObject($order)) {
+            $isValidOrderId = false;
+        }
+
+        $link = $this->context->link->getAdminLink('AdminCustomerThreads', true, [], [
+            'id_customer_thread' => (int) $message['id_customer_thread'],
+            'viewcustomer_thread' => 1,
+        ]);
+
+        $tpl->assign([
+            'thread_url' => Tools::getAdminUrl(basename(_PS_ADMIN_DIR_) . '/' . $link),
+            'link' => $this->context->link,
```

Comment:
Feels weird that `$link` is not assigned to `link`

Reply:
Do you mean `'link' => $link`?

Reply:
Yes. Maybe rename the variable ?

Reply:
I find this confusing:
```
            'thread_url' => Tools::getAdminUrl(basename(_PS_ADMIN_DIR_) . '/' . $link),
            'link' => $this->context->link,
```
I'd expect rather
```
            'thread_url' => $threadUrl,
            'link' => $link,
```
Instead it looks like a mistake not to have `'link' => $link`

Reply:
Ohhhh i see what you mean (I think). :smile: